addons/external_intercom/functions/fnc_connect.sqf

```diff
@@ -0,0 +1,208 @@
+                || !(_headgear isEqualTo (headgear _player))
+                || isNil {_player getVariable "TFAR_ExternalIntercomOutOfRangeVehicle"}
+            }, {
+                params ["_vehicle", "_player", "_intercomMaxRange", "_headgear"];
+
+                if (
+                    alive _vehicle
+                    && _headgear isEqualTo (headgear _player)
+                    && !isNil {_player getVariable "TFAR_ExternalIntercomOutOfRangeVehicle"}
+                ) then {
+                    [_vehicle, _player, [true]] call TFAR_external_intercom_fnc_connect;
+                };
+            }, _this] call CBA_fnc_waitUntilAndExecute;
+        };
+    }, [_vehicle, _player, _intercomMaxRange, _headgear]] call CBA_fnc_waitUntilAndExecute;
+
+} else {
+    private _interactionPointRelative = [_vehicle] call TFAR_external_intercom_fnc_getInteractionPoint;
+
+    // Keep connection alive until out of range or disconnected
+    [{
+        params ["_vehicle", "_player", "_interactionPointRelative"];
+        ((_vehicle modelToWorld _interactionPointRelative) distance _player) > _vehicle getVariable ["TFAR_externalIntercomMaxRange_Phone", TFAR_externalIntercomMaxRange_Phone]
+        || isNil {_player getVariable "TFAR_ExternalIntercomVehicle"}
+    }, {
+        params ["_vehicle", "_player"];
+        [_vehicle, _player] call TFAR_external_intercom_fnc_disconnect;
+    }, [_vehicle, _player, _interactionPointRelative]] call CBA_fnc_waitUntilAndExecute;
+
+    private _handset = createSimpleObject [QPATHTOF(data\TFAR_handset.p3d), _player selectionPosition "head"];
+    _handset attachTo [_player, [-0.14,-0.02,0.02], "head", true];
+    _handset setVectorDirAndUp [[-2.5,0.8,0.25],[-1,-1,1]];
+    private _ropeID = ropeCreate [_vehicle, _interactionPointRelative vectorAdd [0,0.2,0], _handset, "plug", (_vehicle getVariable ["TFAR_externalIntercomMaxRange_Phone", TFAR_externalIntercomMaxRange_Phone]) - 1, ["", [0,0,-1]], ["", [0,0,-1]], "RopeSmallWire"];
+    _vehicle setVariable ["TFAR_ExternalIntercomRopeIDs", [_ropeID, _handset], true];
+
+    // Eventhandlers
+    if (isNil {_vehicle getVariable "TFAR_ExternalIntercomRopeEH"}) then {
+        _vehicle setVariable ["TFAR_ExternalIntercomRopeEH",
+            _vehicle addEventHandler ["RopeBreak", {
+                params ["_vehicle", "_ropeID"];
+                systemChat "Rope broke";
+                systemChat str _this;
+                if ((_vehicle getVariable ["TFAR_ExternalIntercomRopeIDs", [nil, nil]] select 0) isEqualTo _ropeID) then {
+                    [_vehicle, _vehicle getVariable ["TFAR_ExternalIntercomSpeakers", [objNull, []]] select 0] call TFAR_external_intercom_fnc_disconnect;
+                    _vehicle removeEventHandler ["RopeBreak", _thisEventHandler];
+                };
+            }],
+            true
+        ];
+        diag_log "Registered RopeBreak EH";
+        diag_log (_vehicle getVariable "TFAR_ExternalIntercomRopeEH");
```

Comment:
missing cleanup.
You can use TRACE_ macros, and enable debug mode based on `__A3_DEBUG__` macro